source/codegen/templates/proto.mako

```diff
@@ -41,8 +41,10 @@ service ${service_class_prefix} {
 % endfor
 }
 
-% for group_name, attributes in common_helpers.get_attribute_groups(data).items():
-${mako_helper.define_attribute_enum(group_name, attributes)}\
+% for group in common_helpers.get_attribute_groups(data):
+% for data_type, attributes in group.get_attributes_split_by_type().items():
+${mako_helper.define_attribute_enum(group.name, data_type, attributes)}\
+% endfor
 % endfor
 
```

Comment:
Minor spacing fix. To get empty lines between the enums in the proto file can you remove the blank line on line 49 here. Then in `define_attribute_enum` add a blank line after the closing `}` so each enum is separated from the next.